core/src/main/java/org/moditect/commands/GenerateModuleInfo.java

```diff
@@ -187,78 +187,78 @@ private void updateModuleInfo(Map<String, Boolean> optionalityPerModule, ModuleD
         }
 
-        List<ModuleRequiresStmt> requiresStatements = moduleDeclaration.getNodesByType( ModuleRequiresStmt.class );
+        List<ModuleRequiresDirective> requiresStatements = moduleDeclaration.findAll( ModuleRequiresDirective.class );
 
-        for ( ModuleRequiresStmt moduleRequiresStmt : requiresStatements ) {
-            if ( Boolean.TRUE.equals( optionalityPerModule.get( moduleRequiresStmt.getNameAsString() ) ) ) {
-                moduleRequiresStmt.addModifier( Modifier.STATIC );
+        for ( ModuleRequiresDirective moduleRequiresDirective : requiresStatements ) {
+            if ( Boolean.TRUE.equals( optionalityPerModule.get( moduleRequiresDirective.getNameAsString() ) ) ) {
+                moduleRequiresDirective.addModifier( Modifier.Keyword.STATIC );
             }
 
             // update any requires clauses to modules modularized with us to use the assigned module
             // name instead of the automatic module name
             for ( DependencyDescriptor dependency : dependencies ) {
-                if ( dependency.getOriginalModuleName().equals( moduleRequiresStmt.getNameAsString() ) && dependency.getAssignedModuleName() != null ) {
-                    moduleRequiresStmt.setName( dependency.getAssignedModuleName() );
+                if ( dependency.getOriginalModuleName().equals( moduleRequiresDirective.getNameAsString() ) && dependency.getAssignedModuleName() != null ) {
+                    moduleRequiresDirective.setName( dependency.getAssignedModuleName() );
                 }
             }
 
             for ( DependencePattern dependence : requiresPatterns ) {
-                if ( dependence.matches( moduleRequiresStmt.getNameAsString() ) ) {
+                if ( dependence.matches( moduleRequiresDirective.getNameAsString() ) ) {
                     if ( !dependence.isInclusive() ) {
-                        moduleDeclaration.remove( moduleRequiresStmt );
+                        moduleDeclaration.remove( moduleRequiresDirective );
                     }
                     if ( dependence.isMatchAll() && dependence.getModifiers().isEmpty() ) {
-                        moduleRequiresStmt.getModifiers().remove( Modifier.TRANSITIVE );
+                        moduleRequiresDirective.removeModifier(Modifier.Keyword.TRANSITIVE );
                     }
                     else {
-                        moduleRequiresStmt.getModifiers().clear();
+                        moduleRequiresDirective.getModifiers().clear();
                         dependence.getModifiers()
                             .stream()
-                            .map( m -> Modifier.valueOf( m.toUpperCase( Locale.ENGLISH ) ) )
-                            .forEach( m -> moduleRequiresStmt.getModifiers().add( m ) );
+                            .map( m -> Modifier.Keyword.valueOf( m.toUpperCase( Locale.ENGLISH ) ) )
+                            .forEach( m -> moduleRequiresDirective.addModifier(m) );
```

Comment:
There are a bunch of methods on the "NodeWithModifiers" interface for easier modifier manipulation.